tests/test_sorted_results.py

```diff
@@ -43,42 +43,66 @@ def test_empty(self):
             avg_non_gpu_metric_values_list=[avg_non_gpu_metrics],
             comparator=self.result_comparator)
 
-        self.assertTrue(self.result_heap.empty())
-        self.result_heap.add_result(result=result)
-        self.assertFalse(self.result_heap.empty())
-        self.result_heap.next_best_result()
-        self.assertTrue(self.result_heap.empty())
+        self.assertTrue(self.sorted_results.empty())
+        self.sorted_results.add_result(result=result)
+        self.assertFalse(self.sorted_results.empty())
+        self.sorted_results.next_best_result()
+        self.assertTrue(self.sorted_results.empty())
 
     def test_add_results(self):
         avg_gpu_metrics = {0: {'gpu_used_memory': 6000, 'gpu_utilization': 60}}
         avg_non_gpu_metrics = {'perf_throughput': 100, 'perf_latency_p99': 4000}
         for _ in range(10):
-            self.result_heap.add_result(
+            self.sorted_results.add_result(
                 construct_run_config_result(
                     avg_gpu_metric_values=avg_gpu_metrics,
                     avg_non_gpu_metric_values_list=[avg_non_gpu_metrics],
                     comparator=self.result_comparator))
 
-        results = self.result_heap.results()
+        results = self.sorted_results.results()
         self.assertEqual(len(results), 10)
 
+    def test_add_results_same_config(self):
```

Comment:
I'd like to see a test that shows/protects the dynamic sorting.  Have 2 entries where A > B.  Add something to B so that now B > A.

Reply:
Agreed, bad omission on my part. I've added a unit test to cover this case.